minecraft/auth/live.go

```diff
@@ -97,6 +99,33 @@ func RequestLiveTokenWriter(w io.Writer) (*oauth2.Token, error) {
 	panic("unreachable")
 }
 
+var (
+	serverDateMu sync.Mutex
+	// serverDate represents the most recent server date received from Microsoft servers.
+	// It's used for the signed requests which can be blocked if the users device time is not synced.
+	// It uses the date received from the unsigned requests.
+	serverDate time.Time
```

Comment:
Bit of a nitpick, but would "server time" be a better way to describe this, since in Go it's called a time.Time compared to a Date in javascript etc.